# บทนำ: การใช้งาน Jupyter Notebook เบื้องต้น

ในหลักสูตรนี้ เราจะใช้ **Jupyter Notebook** เป็นเครื่องมือหลักสำหรับเรียนรู้ Python และการวิเคราะห์ข้อมูล

Jupyter Notebook ช่วยให้เราสามารถ

- เขียนและรันโค้ด Python ทีละส่วน
- ดูผลลัพธ์ได้ทันที
- เขียนคำอธิบายและบันทึกข้อสังเกต
- จัดเก็บขั้นตอนการวิเคราะห์ไว้ในไฟล์เดียวกัน
- ย้อนกลับมาตรวจสอบและรันกระบวนการเดิมซ้ำได้

สำหรับผู้ที่คุ้นเคยกับ Excel สามารถมอง Notebook เป็น “สมุดวิเคราะห์ข้อมูล” ที่รวมคำอธิบาย ขั้นตอนการคำนวณ และผลลัพธ์ไว้ในที่เดียวกัน

## ผลการเรียนรู้ที่คาดหวัง

เมื่อเรียนจบบทนี้ ผู้เรียนจะสามารถ

1. อธิบายองค์ประกอบและบทบาทของ Jupyter Notebook ได้
2. สร้างและรัน Code Cell และ Markdown Cell ได้
3. ใช้ Python คำนวณผลรวม ค่าเฉลี่ย ค่าสูงสุด และค่าต่ำสุดเบื้องต้นได้
4. จัดลำดับและบันทึกขั้นตอนการวิเคราะห์ใน Notebook ได้

## ลำดับเนื้อหา

บทเรียนนี้ประกอบด้วยหัวข้อต่อไปนี้

1. Jupyter Notebook คืออะไร
2. ส่วนประกอบของ Notebook
3. วิธีการรัน Cell
4. การใช้งาน Code Cell
5. การใช้งาน Markdown Cell
6. การคำนวณพื้นฐานด้วย Python
7. การคำนวณข้อมูลหลายค่า
8. ลำดับการทำงานและ Kernel
9. การอ่าน Error เบื้องต้น
10. การบันทึกผลการวิเคราะห์
11. แบบฝึกหัดท้ายบท

## 1. Jupyter Notebook คืออะไร

**Jupyter Notebook** คือเครื่องมือที่ช่วยให้เราเขียนและรันโค้ด Python แบบทีละส่วนได้

ภายใน Notebook เราสามารถจัดเก็บสิ่งต่อไปนี้ไว้ด้วยกัน

- โค้ด Python
- ผลลัพธ์จากการรันโค้ด
- คำอธิบาย
- ตาราง
- กราฟ
- บันทึกและข้อสรุปจากการวิเคราะห์

ในงานวิเคราะห์ข้อมูล Notebook ช่วยให้เราทำงานอย่างเป็นลำดับ และย้อนกลับมาตรวจสอบได้ว่าเราใช้ข้อมูลอย่างไร คำนวณอย่างไร และได้ข้อสรุปอย่างไร

### เปรียบเทียบ Jupyter Notebook กับ Excel

| งานที่ทำ | Excel | Jupyter Notebook / Python |
|---|---|---|
| คำนวณข้อมูล | เขียนสูตรใน Cell เช่น `=A1+B1` | เขียนคำสั่ง เช่น `100 + 200` |
| ดูผลลัพธ์ | แสดงผลใน Cell | แสดงผลใต้ Code Cell |
| เขียนคำอธิบาย | ใช้ Cell หรือ Comment | ใช้ Markdown Cell |
| ทำงานซ้ำ | Copy หรือ Drag สูตร | รันโค้ดเดิมกับข้อมูลใหม่ |
| วิเคราะห์ข้อมูล | Filter, Sort, PivotTable | ใช้ Python และ pandas |
| ตรวจสอบขั้นตอน | ตรวจสูตรในแต่ละ Cell | อ่าน Notebook จากบนลงล่าง |

## 2. ส่วนประกอบของ Notebook

Notebook ประกอบด้วยช่องสำหรับใส่เนื้อหาที่เรียกว่า **Cell**

Cell ที่ใช้บ่อยมี 2 ประเภทหลัก

1. **Code Cell**  
   ใช้เขียนและรันคำสั่ง Python

2. **Markdown Cell**  
   ใช้เขียนหัวข้อ คำอธิบาย ตาราง หรือบันทึกผลการวิเคราะห์

## 3. วิธีการรัน Cell

การรัน Cell คือการสั่งให้ Notebook ประมวลผลเนื้อหาภายใน Cell นั้น

วิธีที่ใช้บ่อย ได้แก่

- คลิกปุ่ม **Run**
- กด **Shift + Enter** เพื่อรัน Cell และเลื่อนไปยัง Cell ถัดไป
- กด **Ctrl + Enter** เพื่อรัน Cell โดยยังเลือก Cell เดิมไว้

เมื่อรันแล้ว

- Code Cell จะแสดงผลลัพธ์ใต้ Cell
- Markdown Cell จะแสดงข้อความที่จัดรูปแบบแล้ว

## 4. การใช้งาน Code Cell

**Code Cell** ใช้สำหรับเขียนและรันคำสั่ง Python

ทดลองรัน Code Cell ต่อไปนี้

In [23]:
print("Hello, Jupyter Notebook")

Hello, Jupyter Notebook


จากตัวอย่าง

- `print()` เป็นคำสั่งสำหรับแสดงผล
- ข้อความที่อยู่ภายในเครื่องหมาย `" "` คือข้อความที่ต้องการแสดง
- ผลลัพธ์จะแสดงอยู่ใต้ Code Cell หลังจากรัน

In [1]:
print("สวัสดีครับ/ค่ะ")
print("เรากำลังเริ่มเรียน Python สำหรับงานวิเคราะห์ข้อมูล")

สวัสดีครับ/ค่ะ
เรากำลังเริ่มเรียน Python สำหรับงานวิเคราะห์ข้อมูล


Code Cell หนึ่ง Cell สามารถมีคำสั่งได้มากกว่าหนึ่งบรรทัด

Python จะประมวลผลคำสั่งจากบรรทัดบนลงล่าง

## 5. การใช้งาน Markdown Cell

**Markdown Cell** ใช้เขียนข้อความประกอบการวิเคราะห์ โดยไม่ได้ใช้รันคำสั่ง Python

เราสามารถใช้ Markdown Cell เพื่อ

- เขียนชื่อเรื่องและหัวข้อ
- อธิบายขั้นตอนการทำงาน
- บันทึกสมมติฐาน
- สรุปผลการวิเคราะห์
- สร้างรายการและตาราง
- แสดงโค้ดตัวอย่างโดยไม่รันโค้ด

Markdown Cell ช่วยให้ผู้อื่น รวมถึงตัวเราในอนาคต เข้าใจว่า Notebook นี้ใช้ข้อมูลอะไร ทำอะไร และได้ผลลัพธ์อย่างไร

### ตัวอย่างการจัดรูปแบบ Markdown

# หัวข้อระดับที่ 1

## หัวข้อระดับที่ 2

### หัวข้อระดับที่ 3

ข้อความนี้มี **ข้อความตัวหนา** และ *ข้อความตัวเอียง*

รายการแบบ bullet

- โหลดข้อมูล
- ตรวจสอบข้อมูล
- วิเคราะห์ข้อมูล
- สรุปผล

รายการแบบลำดับ

1. เตรียมข้อมูล
2. คำนวณผลลัพธ์
3. ตรวจสอบผล
4. เขียนข้อสรุป

คำสั่ง Python แบบสั้นสามารถเขียนเป็น `print("Hello")` ได้

## 6. การคำนวณพื้นฐานด้วย Python

เราสามารถใช้ Python เป็นเครื่องคิดเลขเพื่อคำนวณค่าพื้นฐานได้

ใน Excel เราอาจเขียน

```excel
=10+20
```

ใน Python เราเขียน


In [25]:
10 + 20

30

### ตัวอย่างการคำนวณยอดขาย

สมมติว่ามีข้อมูลดังนี้

| รายการ | จำนวน | ราคาต่อหน่วย |
|---|---:|---:|
| สินค้า A | 10 | 120 |
| สินค้า B | 5 | 300 |

ยอดขายของแต่ละสินค้าคำนวณจาก

> จำนวน × ราคาต่อหน่วย

In [4]:
quantity_a = 10
price_a = 120

sales_a = quantity_a * price_a

sales_a

1200

In [5]:
quantity_b = 5
price_b = 300

sales_b = quantity_b * price_b

sales_b

1500

In [6]:
total_sales = sales_a + sales_b

total_sales

2700

จากตัวอย่าง เราเริ่มเห็นองค์ประกอบสำคัญของ Python ได้แก่

- การตั้งชื่อให้ข้อมูล เช่น `quantity_a`
- การนำข้อมูลไปคำนวณ
- การเก็บผลลัพธ์ไว้ใช้ต่อ เช่น `sales_a`
- การนำผลลัพธ์หลายค่ามาคำนวณร่วมกัน

ชื่อที่ใช้เก็บข้อมูลเรียกว่า **ตัวแปร** ซึ่งจะเรียนอย่างละเอียดในบทถัดไป

## 7. การคำนวณข้อมูลหลายค่า

ในงานข้อมูล เรามักต้องคำนวณค่าหลายรายการพร้อมกัน

Python สามารถเก็บค่าหลายค่าไว้ภายในเครื่องหมาย `[ ]` ได้ ดังตัวอย่างต่อไปนี้

In [7]:
sales = [1200, 1500, 800, 2200, 1750]

sales

[1200, 1500, 800, 2200, 1750]

ข้อมูลที่อยู่ภายในเครื่องหมาย `[ ]` เรียกว่า **List**

ในบทนี้จะใช้ List เพื่อทดลองคำนวณข้อมูลหลายค่าเท่านั้น ส่วนรายละเอียดของ List จะเรียนในบทต่อไป

### การหาผลรวม

ใน Excel ใช้

```excel
=SUM(A1:A5)
```

ใน Python ใช้ `sum()`

In [9]:
sum(sales)

7450

### การนับจำนวนรายการ

ใน Excel อาจใช้

```excel
=COUNT(A1:A5)
```

ใน Python ใช้ `len()`

In [32]:
len(sales)

5

### การหาค่าเฉลี่ย

ค่าเฉลี่ยคำนวณจาก

> ผลรวม ÷ จำนวนรายการ

In [33]:
average_sales = sum(sales) / len(sales)

average_sales

1490.0

### การหาค่าสูงสุดและค่าต่ำสุด

ใน Excel ใช้

```excel
=MAX(A1:A5)
=MIN(A1:A5)
```

ใน Python ใช้ `max()` และ `min()`

In [ ]:
max(sales)

2200

In [35]:
min(sales)

800

## 8. ลำดับการทำงานใน Notebook

Notebook ควรจัดเนื้อหาให้อ่านจากบนลงล่างตามลำดับต่อไปนี้

1. ระบุวัตถุประสงค์
2. เตรียมหรือกำหนดข้อมูล
3. เขียนโค้ดเพื่อคำนวณหรือวิเคราะห์
4. ตรวจสอบผลลัพธ์
5. เขียนข้อสังเกตหรือข้อสรุป

แนวทางนี้ทำให้ Notebook เป็นทั้งพื้นที่เขียนโค้ดและเอกสารบันทึกกระบวนการวิเคราะห์

### ลำดับการรัน Cell

ลำดับการรัน Cell มีผลต่อผลลัพธ์

ตัวอย่างเช่น ถ้า Cell หนึ่งต้องใช้ตัวแปร `total_sales` เราต้องรัน Cell ที่สร้าง `total_sales` ก่อน มิฉะนั้น Python จะยังไม่รู้จักตัวแปรดังกล่าว

หมายเลขที่แสดงด้านซ้ายของ Code Cell เช่น `[1]`, `[2]` หรือ `[3]` แสดงลำดับที่ Cell ถูกประมวลผล ไม่ใช่ตำแหน่งของ Cell ภายใน Notebook

### Kernel คืออะไร

**Kernel** คือส่วนที่ประมวลผลโค้ดและจดจำค่าต่าง ๆ ที่สร้างขึ้นระหว่างการทำงาน

ข้อควรทราบ

- เมื่อรัน Cell ตัวแปรจะถูกเก็บไว้ใน Kernel
- เมื่อแก้ไขข้อมูล ต้องรัน Cell ที่แก้ไขใหม่
- Cell อื่นที่อาศัยข้อมูลนั้นอาจต้องรันใหม่ด้วย
- เมื่อ Restart Kernel ค่าที่เคยเก็บไว้จะหายไป
- หลัง Restart Kernel ควรรัน Cell ใหม่ตั้งแต่ต้นตามลำดับ

การเลือกคำสั่ง **Restart Kernel and Run All Cells** ช่วยตรวจสอบว่า Notebook สามารถทำงานจากบนลงล่างได้โดยไม่อาศัยค่าที่ตกค้างจากการรันก่อนหน้า

## 9. การอ่าน Error เบื้องต้น

Error เป็นเรื่องปกติในการเขียน Python

เมื่อพบ Error ควรเริ่มจาก

1. อ่านข้อความบรรทัดสุดท้าย
2. ตรวจสอบชื่อคำสั่งหรือตัวแปร
3. ตรวจสอบว่าได้รัน Cell ที่จำเป็นแล้วหรือยัง
4. แก้ไขและทดลองรันใหม่

In [10]:
# ทดลองรัน Cell นี้และสังเกต Error

total_revenue

NameError: name 'total_revenue' is not defined

Code ด้านบนจะเกิด `NameError` เพราะยังไม่มีการสร้างตัวแปรชื่อ `total_revenue`

Error ลักษณะนี้มักเกิดจาก

- พิมพ์ชื่อตัวแปรผิด
- ยังไม่ได้รัน Cell ที่สร้างตัวแปร
- Restart Kernel แล้วไม่ได้รัน Cell ก่อนหน้าใหม่

In [11]:
# สร้างตัวแปรก่อนเรียกใช้งาน

total_revenue = 1000 + 2500 + 1800

total_revenue

5300

หลังจากสร้างตัวแปร `total_revenue` แล้ว Python จึงสามารถแสดงค่าของตัวแปรได้

การพบ Error ไม่ได้หมายความว่าเราไม่สามารถเขียนโปรแกรมได้ แต่เป็นข้อมูลที่ช่วยให้เรารู้ว่าควรตรวจสอบหรือแก้ไขส่วนใด

## 10. การบันทึกผลการวิเคราะห์

การวิเคราะห์ข้อมูลไม่ได้สิ้นสุดเมื่อได้ตัวเลข แต่ควรอธิบายได้ว่า

- ใช้ข้อมูลอะไร
- คำนวณอย่างไร
- ได้ผลลัพธ์อะไร
- ผลลัพธ์มีความหมายอย่างไร
- มีข้อสังเกตหรือข้อจำกัดใด

เราควรใช้ Markdown Cell บันทึกคำอธิบายและข้อสรุปไว้ใกล้กับผลลัพธ์ที่เกี่ยวข้อง

### ตัวอย่างบันทึกผลการวิเคราะห์

จากข้อมูลยอดขาย 5 รายการ พบว่า

- ยอดขายรวมเท่ากับ 7,450 บาท
- ยอดขายเฉลี่ยเท่ากับ 1,490 บาท
- ยอดขายสูงสุดเท่ากับ 2,200 บาท
- ยอดขายต่ำสุดเท่ากับ 800 บาท

ข้อสังเกตเบื้องต้นคือยอดขายแต่ละรายการแตกต่างกันค่อนข้างมาก จึงควรตรวจสอบเพิ่มเติมว่าความแตกต่างดังกล่าวเกี่ยวข้องกับประเภทสินค้า กลุ่มลูกค้า หรือช่วงเวลาขายหรือไม่

# แบบฝึกหัดท้ายบท

แบบฝึกหัดนี้ให้ผู้เรียนทดลองเปลี่ยนงานคำนวณพื้นฐานที่เคยทำใน Excel มาใช้ Python โดยใช้บริบทของข้อมูลเชิงสำรวจในงานสังคมสงเคราะห์

จุดประสงค์ของแบบฝึกหัดคือ

- ฝึกสร้างและรัน Code Cell
- ฝึกอ่านผลลัพธ์
- ฝึกแก้ไขข้อมูลและรันคำสั่งใหม่
- ฝึกบันทึกผลด้วย Markdown Cell

ให้ทดลองทำแบบฝึกหัดด้วยตนเองก่อนเปิดดูเฉลย

## แบบฝึกหัดที่ 1: ใช้ Python แทนเครื่องคิดเลข

ในการเก็บข้อมูลภาคสนาม มีผู้ตอบแบบสำรวจจาก 3 พื้นที่ ดังนี้

- พื้นที่ A มีผู้ตอบแบบสำรวจ 45 คน
- พื้นที่ B มีผู้ตอบแบบสำรวจ 38 คน
- พื้นที่ C มีผู้ตอบแบบสำรวจ 52 คน

ใน Excel เราอาจเขียนสูตร

```excel
=45+38+52
```

ให้เขียน Python เพื่อคำนวณจำนวนผู้ตอบแบบสำรวจรวมทั้งหมด

In [38]:
# เขียนคำตอบของคุณใน Cell นี้

> ### เฉลยแบบฝึกหัดที่ 1
>
> นำจำนวนผู้ตอบจากทั้ง 3 พื้นที่มาบวกกัน

In [12]:
45 + 38 + 52

135

## แบบฝึกหัดที่ 2: คำนวณคะแนนรวมจากแบบสำรวจ

มีข้อมูลคะแนนความต้องการความช่วยเหลือใน 3 ประเด็น โดยแต่ละประเด็นใช้คะแนนตั้งแต่ 1 ถึง 5

| ประเด็นที่สำรวจ | จำนวนผู้ตอบ | คะแนนเฉลี่ย |
|---|---:|---:|
| ความต้องการด้านรายได้ | 40 | 4.2 |
| ความต้องการด้านที่อยู่อาศัย | 35 | 3.8 |
| ความต้องการด้านสุขภาพจิต | 30 | 4.5 |

คะแนนรวมของแต่ละประเด็นคำนวณจาก

> จำนวนผู้ตอบ × คะแนนเฉลี่ย

ให้ใช้ Python คำนวณ

1. คะแนนรวมด้านรายได้
2. คะแนนรวมด้านที่อยู่อาศัย
3. คะแนนรวมด้านสุขภาพจิต
4. คะแนนรวมจากทั้ง 3 ประเด็น

In [13]:
# ข้อมูลตั้งต้น

income_respondents = 40
income_average_score = 4.2

housing_respondents = 35
housing_average_score = 3.8

mental_health_respondents = 30
mental_health_average_score = 4.5

In [40]:
# เขียนคำตอบของคุณใน Cell นี้

> ### เฉลยแบบฝึกหัดที่ 2
>
> คำนวณคะแนนรวมของแต่ละประเด็นก่อน แล้วจึงนำคะแนนทั้ง 3 ประเด็นมารวมกัน

In [14]:
income_total_score = income_respondents * income_average_score
housing_total_score = housing_respondents * housing_average_score
mental_health_total_score = (
    mental_health_respondents * mental_health_average_score
)

overall_total_score = (
    income_total_score
    + housing_total_score
    + mental_health_total_score
)

print("คะแนนรวมด้านรายได้:", income_total_score)
print("คะแนนรวมด้านที่อยู่อาศัย:", housing_total_score)
print("คะแนนรวมด้านสุขภาพจิต:", mental_health_total_score)
print("คะแนนรวมทั้งหมด:", overall_total_score)

คะแนนรวมด้านรายได้: 168.0
คะแนนรวมด้านที่อยู่อาศัย: 133.0
คะแนนรวมด้านสุขภาพจิต: 135.0
คะแนนรวมทั้งหมด: 436.0


## แบบฝึกหัดที่ 3: สรุปข้อมูลหลายค่า

ในการสำรวจครัวเรือนเปราะบาง นักสังคมสงเคราะห์พบจำนวนครัวเรือนที่ต้องการความช่วยเหลือเร่งด่วนใน 6 พื้นที่ ดังนี้

```python
urgent_households = [18, 25, 14, 32, 27, 21]
```

ให้ใช้ Python คำนวณ

จำนวนครัวเรือนรวม
จำนวนครัวเรือนเฉลี่ยต่อพื้นที่
จำนวนครัวเรือนสูงสุด
จำนวนครัวเรือนต่ำสุด

In [16]:
urgent_households = [18, 25, 14, 32, 27, 21]

In [42]:
# เขียนคำตอบของคุณใน Cell นี้

> ### เฉลยแบบฝึกหัดที่ 3
>
> ใช้ `sum()` หาผลรวม ใช้ `len()` นับจำนวนพื้นที่ และใช้ `max()` กับ `min()` หาค่าสูงสุดและค่าต่ำสุด

In [17]:
total_urgent_households = sum(urgent_households)
average_urgent_households = (
    total_urgent_households / len(urgent_households)
)
maximum_urgent_households = max(urgent_households)
minimum_urgent_households = min(urgent_households)

print("จำนวนครัวเรือนรวม:", total_urgent_households)
print("จำนวนครัวเรือนเฉลี่ยต่อพื้นที่:", average_urgent_households)
print("จำนวนครัวเรือนสูงสุด:", maximum_urgent_households)
print("จำนวนครัวเรือนต่ำสุด:", minimum_urgent_households)

จำนวนครัวเรือนรวม: 137
จำนวนครัวเรือนเฉลี่ยต่อพื้นที่: 22.833333333333332
จำนวนครัวเรือนสูงสุด: 32
จำนวนครัวเรือนต่ำสุด: 14


## แบบฝึกหัดที่ 4: บันทึกผลการวิเคราะห์ด้วย Markdown

จากผลลัพธ์ในแบบฝึกหัดที่ 3 ให้สร้าง Markdown Cell ใหม่ แล้วเขียนสรุปผลโดยระบุ

- จำนวนครัวเรือนรวม
- จำนวนครัวเรือนเฉลี่ยต่อพื้นที่
- จำนวนครัวเรือนสูงสุด
- จำนวนครัวเรือนต่ำสุด
- ข้อสังเกตเบื้องต้นอย่างน้อย 1 ข้อ

สามารถใช้รูปแบบต่อไปนี้เป็นแนวทาง

```markdown
จากข้อมูลจำนวนครัวเรือนที่ต้องการความช่วยเหลือเร่งด่วน พบว่า

- จำนวนครัวเรือนรวมเท่ากับ ...
- จำนวนครัวเรือนเฉลี่ยต่อพื้นที่เท่ากับ ...
- จำนวนครัวเรือนสูงสุดเท่ากับ ...
- จำนวนครัวเรือนต่ำสุดเท่ากับ ...

ข้อสังเกตเบื้องต้นคือ ...
```

-- เขียนคำตอบของคุณใน Markdown Cell นี้ --

> ### เฉลยแบบฝึกหัดที่ 4
>
> จากข้อมูลจำนวนครัวเรือนที่ต้องการความช่วยเหลือเร่งด่วนใน 6 พื้นที่ พบว่า
>
> - มีครัวเรือนที่ต้องการความช่วยเหลือรวม 137 ครัวเรือน
> - แต่ละพื้นที่มีครัวเรือนที่ต้องการความช่วยเหลือเฉลี่ยประมาณ 22.83 ครัวเรือน
> - พื้นที่ที่มีจำนวนครัวเรือนสูงสุดมี 32 ครัวเรือน
> - พื้นที่ที่มีจำนวนครัวเรือนต่ำสุดมี 14 ครัวเรือน
>
> ข้อสังเกตเบื้องต้นคือจำนวนครัวเรือนของแต่ละพื้นที่แตกต่างกัน จึงควรตรวจสอบเพิ่มเติมว่าพื้นที่ที่มีจำนวนสูงอาจต้องได้รับการจัดสรรบุคลากรหรืองบประมาณมากกว่าพื้นที่อื่นหรือไม่

## แบบฝึกหัดที่ 5: แก้ไขข้อมูลและรัน Cell ใหม่

ให้ทดลองทำตามขั้นตอนต่อไปนี้

1. เพิ่มข้อมูลพื้นที่ที่ 7 ซึ่งมีครัวเรือนที่ต้องการความช่วยเหลือเร่งด่วน 24 ครัวเรือน
2. รัน Cell ที่กำหนดข้อมูลใหม่
3. รันคำสั่งคำนวณผลรวม ค่าเฉลี่ย ค่าสูงสุด และค่าต่ำสุดอีกครั้ง
4. เปรียบเทียบผลลัพธ์ก่อนและหลังเพิ่มข้อมูล

ตอบคำถามต่อไปนี้

- จำนวนครัวเรือนรวมเปลี่ยนไปอย่างไร
- ค่าเฉลี่ยเปลี่ยนไปอย่างไร
- ค่าสูงสุดเปลี่ยนหรือไม่
- ค่าต่ำสุดเปลี่ยนหรือไม่
- การทำงานนี้คล้ายกับการแก้ข้อมูลใน Excel แล้วให้สูตรคำนวณใหม่อย่างไร

In [18]:
# เพิ่มข้อมูลพื้นที่ที่ 7

urgent_households = [18, 25, 14, 32, 27, 21, 24]

In [ ]:
# รันคำสั่งคำนวณใหม่อีกครั้งใน Cell นี้

> ### เฉลยแบบฝึกหัดที่ 5
>
> หลังเพิ่มข้อมูลของพื้นที่ที่ 7 ต้องคำนวณค่าต่าง ๆ ใหม่จากข้อมูลชุดล่าสุด

In [19]:
total_urgent_households = sum(urgent_households)
average_urgent_households = (
    total_urgent_households / len(urgent_households)
)
maximum_urgent_households = max(urgent_households)
minimum_urgent_households = min(urgent_households)

print("จำนวนครัวเรือนรวม:", total_urgent_households)
print("จำนวนครัวเรือนเฉลี่ยต่อพื้นที่:", average_urgent_households)
print("จำนวนครัวเรือนสูงสุด:", maximum_urgent_households)
print("จำนวนครัวเรือนต่ำสุด:", minimum_urgent_households)

จำนวนครัวเรือนรวม: 161
จำนวนครัวเรือนเฉลี่ยต่อพื้นที่: 23.0
จำนวนครัวเรือนสูงสุด: 32
จำนวนครัวเรือนต่ำสุด: 14


> ### การเปรียบเทียบผลลัพธ์
>
> - จำนวนครัวเรือนรวมเพิ่มจาก 137 เป็น 161 ครัวเรือน
> - ค่าเฉลี่ยเพิ่มจากประมาณ 22.83 เป็น 23 ครัวเรือนต่อพื้นที่
> - ค่าสูงสุดยังคงเป็น 32 ครัวเรือน
> - ค่าต่ำสุดยังคงเป็น 14 ครัวเรือน
>
> ผลรวมและค่าเฉลี่ยเปลี่ยน เพราะข้อมูลใหม่ถูกนำมารวมในการคำนวณ ส่วนค่าสูงสุดและค่าต่ำสุดไม่เปลี่ยน เพราะค่าใหม่คือ 24 ซึ่งไม่ได้สูงกว่า 32 หรือต่ำกว่า 14
>
> การทำงานนี้คล้ายกับการเพิ่มข้อมูลใน Excel แล้วคำนวณสูตรใหม่ แต่ใน Notebook ผู้ใช้ต้องรัน Cell ที่แก้ไขและ Cell ที่ใช้คำนวณอีกครั้ง เพื่อให้ผลลัพธ์อ้างอิงข้อมูลล่าสุด